### PR TITLE
Fix Jexl Expressions

### PR DESCRIPTION
**Pull request by AutomatorGuy123**
**Note**:  This only fixes the Jexl Expressions for GUI.  Investigation for API still needs to be done.

### CODE REVIEW

**Comment by AutomatorGuy123** [comment]
The initial fix for the GUI also fixed API.  However, I missed added crypto support for API.  So, in the 2nd commit I added this and the unit test for API & Jexl Expressions.

File: automation-tests/src/main/java/com/automation/common/api/domainObjects/JsonIpDO.java [hunk 1]
```diff
@@ -73,13 +72,13 @@ private void validateResponse() {
             String json = ip.getEntityAsString();
             Map<String, Object> actual = WsUtils.toMap(json);
             String actualIP = String.valueOf(actual.get("ip"));
-            String actualAbout = String.valueOf(actual.get("about"));
-            String actualPro = String.valueOf(actual.get("Pro!"));
+            String actualAbout = String.valueOf(actual.get("geo-ip"));
```
Comment (dn0000001) [inline automation-tests/src/main/java/com/automation/common/api/domainObjects/JsonIpDO.java]: Should be actualGeoIp

File: automation-tests/src/main/java/com/automation/common/api/domainObjects/JsonIpDO.java [hunk 2]
```diff
@@ -73,13 +72,13 @@ private void validateResponse() {
...
+            String actualPro = String.valueOf(actual.get("API Help"));
```
Comment (dn0000001) [inline automation-tests/src/main/java/com/automation/common/api/domainObjects/JsonIpDO.java]: Should be actualApiHelp

File: taf/src/main/java/com/taf/automation/ui/support/DataInstillerUtils.java [hunk 3]
```diff
@@ -107,8 +95,8 @@ public static XStream getXStream(XStream existingXStream) {
      * @param existingConverters - Existing converters to be registered
      * @return XStream
      */
-    public static XStream getXStream(XStream existingXStream, List<DataValueConverter> existingConverters) {
-        XStream xstream = (existingXStream == null) ? getDefaultXStream() : existingXStream;
+    public static XStream getXStream(XStream existingXStream, List<DataValueConverter> existingConverters, JexlContext jexlContext) {
```
Comment (dn0000001) [inline taf/src/main/java/com/taf/automation/ui/support/DataInstillerUtils.java]: Update the JavaDoc for jexlContext

File: taf/src/main/java/com/taf/automation/ui/support/DataInstillerUtils.java [hunk 4]
```diff
@@ -121,10 +109,12 @@ public static XStream getXStream(XStream existingXStream, List<DataValueConverte
      *
      * @return XStream
      */
-    private static XStream getDefaultXStream() {
+    private static XStream getDefaultXStream(JexlContext existingJexlContext) {
```
Comment (dn0000001) [inline taf/src/main/java/com/taf/automation/ui/support/DataInstillerUtils.java]: Update the JavaDoc for jexlContext

File: taf/src/main/java/com/taf/automation/ui/support/DataInstillerUtils.java [hunk 5]
```diff
@@ -225,8 +212,8 @@ private static void registerConverters(XStream xstream, List<DataValueConverter>
      *
      * @return DataGenerator
      */
-    public static DataGenerator getGenerator() {
-        DataGenerator generator = new DataGenerator(getXStream(), getConverters());
+    public static DataGenerator getGenerator(JexlContext jexlContext) {
```
Comment (dn0000001) [inline taf/src/main/java/com/taf/automation/ui/support/DataInstillerUtils.java]: Update the JavaDoc for jexlContext

**Pull request merged**